internal/middleware/cache.go

```diff
@@ -0,0 +1,99 @@
+package middleware
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/auth"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response"
+	"github.com/dilyara4949/flight-booking-api/internal/handler/response/pagination"
+	"github.com/gin-gonic/gin"
+	"github.com/redis/go-redis/v9"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+)
+
+func Cache(cache *redis.Client, ttl time.Duration) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		if !(c.Request.Method == http.MethodGet || c.Request.Method == http.MethodPut) {
```

Comment:
Why do we need to cache `PUT` requests?

Reply:
may be I understood you incorrectly, but previously you mentioned to invalidate cache if update was on existing cache key

Reply:
yes, you need to invalidate cache on edit, but I thought, that it wouldn't be in the same middleware. That works.